pinot-core/src/main/java/org/apache/pinot/core/common/IntermediateStageBlockValSet.java

```diff
@@ -0,0 +1,250 @@
+  }
+
+  @Override
+  public long[] getLongValuesSV() {
+    if (_values == null) {
+      return null;
+    }
+
+    int length = _values.size();
+    long[] values = new long[length];
+    for (int i = 0; i < length; i++) {
+      Object value = _values.get(i);
+      if (value != null) {
+        values[i] = _pinotDataType.toLong(value);
+      }
+    }
+
+    return values;
+  }
+
+  @Override
+  public float[] getFloatValuesSV() {
+    if (_values == null) {
+      return null;
+    }
+    int length = _values.size();
+    float[] values = new float[length];
+    for (int i = 0; i < length; i++) {
+      Object value = _values.get(i);
+      if (value != null) {
+        values[i] = _pinotDataType.toFloat(value);
+      }
+    }
+
+    return values;
+  }
+
+  @Override
+  public double[] getDoubleValuesSV() {
+    if (_values == null) {
+      return null;
+    }
+    int length = _values.size();
+    double[] values = new double[length];
+    for (int i = 0; i < length; i++) {
+      Object value = _values.get(i);
+      if (value != null) {
+        values[i] = _pinotDataType.toDouble(value);
+      }
+    }
+
+    return values;
+  }
+
+  @Override
+  public BigDecimal[] getBigDecimalValuesSV() {
+    if (_values == null) {
+      return null;
+    }
+    int length = _values.size();
+    BigDecimal[] values = new BigDecimal[length];
+    for (int i = 0; i < length; i++) {
+      Object value = _values.get(i);
+      if (value != null) {
+        values[i] = _pinotDataType.toBigDecimal(value);
+      }
+    }
+    return values;
+  }
+
+  @Override
+  public String[] getStringValuesSV() {
+    if (_values == null) {
+      return null;
+    }
+    int length = _values.size();
+    String[] values = new String[length];
+    for (int i = 0; i < length; i++) {
+      Object value = _values.get(i);
+      if (value != null) {
+        values[i] = _pinotDataType.toString(value);
+      }
+    }
+    return values;
+  }
+
+  @Override
+  public byte[][] getBytesValuesSV() {
+    if (_values == null) {
+      return null;
+    }
+    int length = _values.size();
+    byte[][] values = new byte[length][];
+    for (int i = 0; i < length; i++) {
+      Object value = _values.get(i);
+      if (value != null) {
+        values[i] = _pinotDataType.toBytes(value);
+      }
+    }
+    return values;
+  }
+
+  @Override
+  public int[][] getDictionaryIdsMV() {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public int[][] getIntValuesMV() {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public long[][] getLongValuesMV() {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public float[][] getFloatValuesMV() {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public double[][] getDoubleValuesMV() {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public String[][] getStringValuesMV() {
+    throw new UnsupportedOperationException();
```

Comment:
Hi, 
Thanks for putting this work. I was waiting for this as I think it helps me with supporting the [json scalar functions](https://github.com/apache/pinot/pull/10790) in multistage. 

Just a question, why we are not supporting MV types? With the MV types support we can handle scalar functions like `string split` and many other.

Reply:
We currently don't support MV types in Multistage engine - https://github.com/apache/pinot/issues/10658. Eventually when we support MV columns in Multistage and add the corresponding aggregation functions, we'll have to support MV types here.